addons/mail/static/src/core/common/composer.js

```diff
@@ -600,4 +613,31 @@ export class Composer extends Component {
             this.threadService.markAsRead(this.props.composer.thread);
         }
     }
+
+    keepContent() {
+        if (!this.env.inChatter || !this.thread) {
+            return;
+        }
+        const fullComposerContent = document
+            .querySelector(".o_mail_composer_form_view #body_0")
+            ?.innerText.replace(/(\t|\n)+/g, "\n");
+        if (fullComposerContent) {
+            browser.localStorage.setItem(
+                `${this.thread.localId}_fullComposerText`,
+                fullComposerContent
+            );
+        }
+    }
+
+    setComposerInput() {
+        if (!this.env.inChatter || !this.thread) {
+            return;
+        }
```

Comment:
the code would read better if the condition was outside of the method, before calling it